zcl-builtin/matter/data-model/chip/test-cluster.xml

```diff
@@ -83,6 +83,10 @@ limitations under the License.
     <item name="a" type="NestedStructList" array="true" optional="false"/>
   </struct>
 
+  <struct name="DeepNestedStructList">
+    <cluster code="0xFFF1FC05"/>
+    <item name="a" type="NestedStructList" optional="false"/>
```

Comment:
Could you add a test for the helper which will be taking this into account? If zcl_structs then add a generation test for that.

Reply:
The issue is that I could not find any test which was verifying the functionality of `struct_contains_array` - the best I found was 

My thought would be to modify said test to also check for transitive structures - can you point me to said test?

(see also https://github.com/project-chip/zap/pull/1501#issuecomment-2569920792)

Reply:
Thanks for the pointer! Test added